### DISCUSSION TITLE
Possible ease of life additions

### DISCUSSION THREAD

**Discussion by ItsSour** [PartialVolume/shredos.x86_64#92] [Ideas]
Afternoon! I love this program and is certainly the best wiping software I have used, and I can see there are still additions being made which is good to see. I have a few ease of life things that would be nice to have. Let me know your thoughts and keep up the great work!

It would be a nice ease of life addition to have ShredOS save the method, blanking pass, verification pass and PRNG configurations from the previous wipe, so upon a reboot or something of the like, you don't have to go and change all of the configurations again. 

Another thing that would be nice is to be able to have different wipe methods being used for different drives, for example, you have an SSD you want just a 0 pass on and an HDD you want the DoD wipe on, being able to have both running with different methods at the same time would be a nice ease of life. 

I was also thinking a way to add semi custom wipe methods, for example, the place I work, I am required to wipe a hard drive with at least 2 passes, ideally I would love to just do 2 (personally I believe 1 is more than enough anyways), so being able to have a custom method writing ones then writing zeros would be nice (obviously you could add whatever you please, mixing different passes from different methods)

I believe the configuration save addition would be simple enough, but I'm not sure about the other two. Any thoughts?

**Comment by PartialVolume** [comment]
>It would be a nice ease of life addition to have ShredOS save the method, blanking pass, verification pass and PRNG configurations from the previous wipe, so upon a reboot or something of the like, you don't have to go and change all of the configurations again.

That would be pretty quick to implement. If you open an issue with this suggestion I'll mark it as an enhancement. That way I won't forget about it.

>Another thing that would be nice is to be able to have different wipe methods being used for different drives, for example, you have an SSD you want just a 0 pass on and an HDD you want the DoD wipe on, being able to have both running with different methods at the same time would be a nice ease of life.

I'm currently working on the SSD secure erase method and have been thinking the same thing. One of my laptops has a SSD and HDD and being able to wipe both drives simultaneously with different methods would be real useful. Same as above open as an issue and I'll mark it as an enhancement.

>I was also thinking a way to add semi custom wipe methods, for example, the place I work, I am required to wipe a hard drive with at least 2 passes, ideally I would love to just do 2 (personally I believe 1 is more than enough anyways), so being able to have a custom method writing ones then writing zeros would be nice (obviously you could add whatever you please, mixing different passes from different methods)

Also a good idea, I think this or something similar is already open as an enhancement request in ShredOS or nwipe.

**Reply by ItsSour** [reply]
Will do, thanks!
